controller/linodemachine_controller.go

```diff
@@ -429,6 +425,25 @@ func (r *LinodeMachineReconciler) addMachineToLB(
 	return nil
 }
 
+func (r *LinodeMachineReconciler) removeMachineFromLB(
+	ctx context.Context,
+	logger logr.Logger,
+	machineScope *scope.MachineScope,
+) error {
+	if machineScope.LinodeCluster.Spec.Network.LoadBalancerType == "NodeBalancer" {
+		if err := services.DeleteNodeFromNB(ctx, logger, machineScope); err != nil {
+			logger.Error(err, "Failed to remove node from Node Balancer backend")
+			return err
+		}
+	} else if machineScope.LinodeCluster.Spec.Network.LoadBalancerType == "dns" {
```

Comment:
`"NodeBalancer"` and `"dns"` should be constants.

Reply:
While we're on it, it might be worth changing this to be "DNS"

Reply:
I have an upcoming PR to add webhooks for the DNS optional fields. I will make the uppercase change there if that's ok